### PR TITLE
FindTBB.cmake: only search TBB in Config mode if version >= 2019 update 5

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

Avoid the following error when TBB was already found a few lines earlier, by adding a condition `NOT TBB_FOUND` before `find_package(TBB)` :
```
CMake Error at C:/dev/tbb2019_20181203oss_win/tbb2019_20181203oss/cmake/TBBConfig.cmake:83 (add_library):
  add_library cannot create imported target "TBB::tbb" because another target
  with the same name already exists.
```

## Release Management

* Affected package(s): Polyhedron (demo)
* License and copyright ownership: unchanged

### CODE REVIEW

**Comment by lrineau** [comment]
I think we should fix https://github.com/CGAL/cgal/pull/5687, instead. That is a bug from TBB before version 2019 update 5, and we have an easy workaround: https://github.com/CGAL/cgal/pull/5693#discussion_r640631731.

**Review by lrineau — APPROVED** [review]

**Comment by janetournois** [comment]
Be careful : CMake is fixed but the demo does not compile yet with this fix

**Comment by lrineau** [comment]
> Be careful : CMake is fixed but the demo does not compile yet with this fix

The compilation error is:
```
.../Mesh_3/include/CGAL/Mesh_complex_3_in_triangulation_3.h:105:3: error: use of deleted function 'tbb::detail::d1::tbb_hash_compare<std::pair<CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Mesh_vertex_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, std::tuple<int, int, int, int>, int, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Mesh_vertex_generator_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, std::tuple<int, int, int, int>, int, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Triangulation_ds_vertex_base_3<void> > >, CGAL::Triangulation_cell_base_with_info_3<int, CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Compact_mesh_cell_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Polyhedral_mesh_domain_with_features_3<CGAL::Epick, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >, CGAL::Default, int> > >, CGAL::Sequential_tag> > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false>, CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Mesh_vertex_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, std::tuple<int, int, int, int>, int, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Triangulation_ds_vertex_base_3<CGAL::Triangulation_data_structure_3<CGAL::Mesh_vertex_generator_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, std::tuple<int, int, int, int>, int, CGAL::Regular_triangulation_vertex_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Triangulation_ds_vertex_base_3<void> > >, CGAL::Triangulation_cell_base_with_info_3<int, CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Compact_mesh_cell_base_3<CGAL::Robust_weighted_circumcenter_filtered_traits_3<CGAL::Mesh_3::Robust_intersection_traits_3_new<CGAL::Epick> >, CGAL::Polyhedral_mesh_domain_with_features_3<CGAL::Epick, CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epick> >, CGAL::Default, int> > >, CGAL::Sequential_tag> > > >, CGAL::Default, CGAL::Default, CGAL::Default>, false> > >::~tbb_hash_compare()'
```

That is because we have
```c++
  typedef tbb::concurrent_hash_map<Pair_of_vertices, int> Edge_facet_counter;
```
where `Pair_of_vertices` is a `std::pair` of `Vertex_handle`. For TBB before oneAPI, we define an overload of `tbb_hasher`, that can be found by ADL:
https://github.com/CGAL/cgal/blob/daf70dbcf6d400cbece30ae173e3e951e48aed0a/Mesh_3/include/CGAL/Mesh_3/Mesh_complex_3_in_triangulation_3_base.h#L60

But oneAPI-TBB has switched to the use of `std::hash`.

In think we should pass explicitly a `HashCompare` to the [`concurrent_hash_map`](https://spec.oneapi.com/versions/0.5.0/oneTBB/containers/concurrent_hash_map_cls.html). That API is the same for TBB and oneAPI-TBB.

In TBB, the default argument for `HashCompare` was:
```c++
template<typename Key>
struct tbb_hash_compare {
    static size_t hash( const Key& a ) { return tbb_hasher(a); }
    static bool equal( const Key& a, const Key& b ) { return a == b; }
};
```
(see the ADL call to `tbb_hasher`).

In oneAPI-TBB, that is:
```c++
template <typename Key>
class tbb_hash_compare {
public:
    std::size_t hash( const Key& a ) const { return my_hash_func(a); }
    bool equal( const Key& a, const Key& b ) const { return my_key_equal(a, b); }
private:
    std::hash<Key> my_hash_func;
    std::equal_to<Key> my_key_equal;
};
```
(see the use of `std::hash`).

File: Installation/cmake/modules/FindTBB.cmake [hunk 1]
```diff
@@ -188,12 +188,20 @@ endmacro()
 #=============================================================================
 #  Now to actually find TBB
 #
+#start with looking for TBB_DIR and TBB_ROOT
+if((TBB_ROOT OR ENV_TBB_ROOT OR ENV{TBB_DIR} ) AND NOT TBB_FOUND)
```
Comment (lrineau) [inline Installation/cmake/modules/FindTBB.cmake:192]:
```suggestion
if((TBB_ROOT OR ENV{TBB_ROOT} OR ENV{TBB_DIR} ) AND NOT TBB_FOUND)
```

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-138.shtml

**Pull request merged**